contracts/blade/staking/StakeManager.sol

```diff
@@ -24,6 +26,7 @@ contract StakeManager is IStakeManager, Initializable, Ownable2StepUpgradeable,
     bytes32 public domain;
 
     mapping(address => Validator) public validators;
+    EnumerableSet.AddressSet private activeValidatorsSet;
```

Comment:
I dont think introducing a new set adds any value. You already have a map containing all the validators, where its IsActive flag is updated when he registers and unstakes.

What you need to do is just use the getValidators function on the client and filter them by the isActive flag, or you can change your getActiveValidators function on the contract to the filtering, so you dont have to do it on the client.